spec/kobo_bluetooth_spec.lua

```diff
@@ -463,6 +463,415 @@ describe("KoboBluetooth", function()
+            instance:initWithPlugin(mock_plugin)
+            mock_plugin.settings.enable_auto_detection_polling = true
+
+            -- Mock device list
+            instance.device_manager.paired_devices_cache = {
+                {
+                    address = "00:11:22:33:44:55",
+                    name = "Test Device",
+                    connected = true,
+                    paired = true,
+                    trusted = true,
+                },
+            }
+            instance.device_manager.loadPairedDevices = function(self) end
+
+            UIManager:_reset()
+            instance:startAutoDetectionPolling()
+
+            assert.is_not_nil(instance.auto_detection_poll_task)
+            assert.are.equal(1, #UIManager._scheduled_tasks)
+            assert.are.equal(instance.auto_detection_poll_interval, UIManager._scheduled_tasks[1].time)
+        end)
+
+        it("should not start polling when setting is disabled", function()
+            setMockPopenOutput("variant boolean true")
+
+            local instance = KoboBluetooth:new()
+            instance:initWithPlugin(mock_plugin)
+            mock_plugin.settings.enable_auto_detection_polling = false
+
+            UIManager:_reset()
+            instance:startAutoDetectionPolling()
+
+            assert.is_nil(instance.auto_detection_poll_task)
+            assert.are.equal(0, #UIManager._scheduled_tasks)
+        end)
+
+        it("should not start polling when already running", function()
+            setMockPopenOutput("variant boolean true")
+
+            local instance = KoboBluetooth:new()
+            instance:initWithPlugin(mock_plugin)
+            mock_plugin.settings.enable_auto_detection_polling = true
+
+            instance.device_manager.paired_devices_cache = {}
+            instance.device_manager.loadPairedDevices = function(self) end
+
+            UIManager:_reset()
+            instance:startAutoDetectionPolling()
+            local first_task = instance.auto_detection_poll_task
+
+            instance:startAutoDetectionPolling()
+
+            assert.are.equal(first_task, instance.auto_detection_poll_task)
+            assert.are.equal(1, #UIManager._scheduled_tasks)
+        end)
+
+        it("should stop polling when stopAutoDetectionPolling is called", function()
+            setMockPopenOutput("variant boolean true")
+
+            local instance = KoboBluetooth:new()
+            instance:initWithPlugin(mock_plugin)
+            mock_plugin.settings.enable_auto_detection_polling = true
+
+            instance.device_manager.paired_devices_cache = {}
+            instance.device_manager.loadPairedDevices = function(self) end
+
+            UIManager:_reset()
+            instance:startAutoDetectionPolling()
+            assert.is_not_nil(instance.auto_detection_poll_task)
+
+            instance:stopAutoDetectionPolling()
+
+            assert.is_nil(instance.auto_detection_poll_task)
+        end)
```

Comment:
This test should also confirm that the correct calls are made to remove the task from UIManager